Eliminate tar_options, to work around bugs in new salt version

justinsb opened the pull request:
There appear to be two relevant bugs in the latest salt version (v2014.7.1):

https://github.com/saltstack/salt/issues/20195 means we cannot specify x any more

https://github.com/saltstack/salt/issues/20201 is more serious - it double quotes the filename, meaning the filename cannot be found.  By removing tar_options entirely we go through a different code path, using a Python tarfile extractor, which works.

Fixes issue #3932

satnam6502 commented:
@jbeda : please can you review this? If not let me know, and I shall find someone else. Thanks.

satnam6502 commented:
I am guessing P2 for this, but let me know if you think this is really a P1 issue.

jbeda commented:
We should verify that this doesn't break GCE that is version locked to salt 2014.1.13.  (that version locking is done here: https://github.com/GoogleCloudPlatform/kubernetes/blob/master/cluster/gce/templates/common.sh#L35.

If we can verify this, we can upgrade to the new version of salt there.

jbeda commented:
I'll run the GCE e2e tests with this change.

justinsb commented:
Maybe we should lock AWS to the same version of salt instead?  It doesn't really make sense to have to deal with different salt versions, does it?

jbeda commented:
Well, let's update to 2014.7.1 and go from there.

jbeda commented:
Yeah -- this breaks with the old version of salt.  Let me research what we should do.

jbeda commented:
@justinsb -- I see that https://github.com/jbeda/kubernetes/commit/0d281e4bcf59a1870228a2ce77732b2f4309b42c works for me.  Can you try this and see if it works for you?

Let's get this fixed for you and then look at moving the GCE scripts to 2014.7.1.

jbeda commented:
Fixed typo -- see 183ecfdc9f3bb76582f177dcb1b96649bab47458 instead.

justinsb committed:
Work-around for archive bug in SaltStack 2014.7.1

SaltStack 2014.7.1 double-quotes tar filenames which means the file can't
be found.  A workaround is not to specify tar_options, which hits a different
code path.  But that breaks the older version of SaltStack that GCE is locked
to, hence we need the version conditional (thanks @jbeda)

Fixes issue #3932

justinsb commented:
Confirmed that https://github.com/GoogleCloudPlatform/kubernetes/commit/183ecfdc9f3bb76582f177dcb1b96649bab47458 works for me on AWS with the latest saltstack.

I do think we should all be on the same version of salt, but this at least unbreaks platforms on the latest salt.

jbeda commented:
@justinb -- go ahead and update this PR with that `if` and I'll merge it.

justinsb commented:
Updated with your fix @jbeda - thanks.

SaltStack 2014.7.2 should fix the issue and is due out Feb 6, if not before: https://github.com/saltstack/salt/issues/20201#issuecomment-72124525

Do we want to change the test to != 2014.7.1 ?

jbeda commented:
@justinsb -- I'm going to merge this but feel free to submit new PRs as things develop.

I also filed #3979 to track the version GCE is locked to.  If you want to version lock the AWS version, feel free to draw from the same GCS buckets.
